transferui/src/main/java/com/hyperwallet/android/ui/transfer/viewmodel/CreateTransferViewModel.java

```diff
@@ -0,0 +1,338 @@
+        Transfer transfer = mTransferAvailableFunds.getValue() ?
+                new Transfer.Builder()
+                        .clientTransferID(CLIENT_IDENTIFICATION_PREFIX + UUID.randomUUID().toString())
+                        .sourceToken(mQuoteAvailableFunds.getValue().getSourceToken())
+                        .destinationToken(mTransferDestination.getValue().getField(TOKEN))
+                        .destinationCurrency(mTransferDestination.getValue().getField(TRANSFER_METHOD_CURRENCY))
+                        .notes(mTransferNotes.getValue())
+                        .build() :
+                new Transfer.Builder()
+                        .clientTransferID(CLIENT_IDENTIFICATION_PREFIX + UUID.randomUUID().toString())
+                        .sourceToken(mQuoteAvailableFunds.getValue().getSourceToken())
+                        .destinationToken(mTransferDestination.getValue().getField(TOKEN))
+                        .destinationCurrency(mTransferDestination.getValue().getField(TRANSFER_METHOD_CURRENCY))
+                        .notes(mTransferNotes.getValue())
+                        .destinationAmount(mTransferAmount.getValue())
+                        .build();
+
+        mTransferRepository.createTransfer(transfer, new TransferRepository.CreateTransferCallback() {
+            @Override
+            public void onTransferCreated(@Nullable Transfer transfer) {
+                mQuoteTransfer.postValue(transfer);
+                mIsCreateQuoteLoading.postValue(Boolean.FALSE);
+            }
+
+            @Override
+            public void onError(@NonNull HyperwalletErrors errors) {
+                mQuoteErrors.postValue(new Event<>(errors));
+                mIsCreateQuoteLoading.postValue(Boolean.FALSE);
+            }
+        });
+    }
+
+    @Override
+    protected void onCleared() {
+        super.onCleared();
+        mLoadErrors.removeObserver(mLoadErrorEventObserver);
+        mQuoteErrors.removeObserver(mQuoteEventObserver);
+    }
+
+    private void createLoadErrorObserver() {
+        mLoadErrorEventObserver = new Observer<Event<HyperwalletErrors>>() {
+            @Override
+            public void onChanged(Event<HyperwalletErrors> event) {
+                mLoadErrorEvent.postValue(event);
+            }
+        };
+        mLoadErrors.observeForever(mLoadErrorEventObserver);
+    }
+
+    private void createQuoteErrorObserver() {
+        mQuoteEventObserver = new Observer<Event<HyperwalletErrors>>() {
+            @Override
+            public void onChanged(Event<HyperwalletErrors> event) {
+                mQuoteErrorEvent.postValue(event);
+            }
+        };
+        mQuoteErrors.observeForever(mQuoteEventObserver);
+    }
+
+    private void loadTransferSource() {
+        mUserRepository.loadUser(new UserRepository.LoadUserCallback() {
+            @Override
+            public void onUserLoaded(@NonNull HyperwalletUser user) {
+                loadTransferDestination(user.getToken());
```

Comment:
heads up, we will need a member variable to hold the user token value - (error handling/retry use cases) - I will leave it up to you to handle it now or in the future sprints